### PR TITLE
Misc changes from eip-4844 branch

### PR DESCRIPTION
**Pull request by dapplion**
**Motivation**

- Ongoing PR https://github.com/ChainSafe/lodestar/pull/4774 includes multiple small diffs found throughout multiple weeks.
 
They are valuable but barely deserve individual PRs. Feel free to :-1: if the scope of this PR is too wide and should be split or re-done

**Description**

Misc changes from eip-4844 branch

### CODE REVIEW

File: packages/beacon-node/src/chain/chain.ts [hunk 1]
```diff
@@ -353,11 +353,11 @@ export class BeaconChain implements IBeaconChain {
     return await this.db.block.get(fromHexString(block.blockRoot));
   }
 
-  async produceBlock(blockAttributes: BlockAttributes): Promise<allForks.BeaconBlock> {
+  produceBlock(blockAttributes: BlockAttributes): Promise<allForks.BeaconBlock> {
     return this.produceBlockWrapper<BlockType.Full>(BlockType.Full, blockAttributes);
   }
 
-  async produceBlindedBlock(blockAttributes: BlockAttributes): Promise<allForks.BlindedBeaconBlock> {
+  produceBlindedBlock(blockAttributes: BlockAttributes): Promise<allForks.BlindedBeaconBlock> {
```
Comment (dapplion) [inline packages/beacon-node/src/chain/chain.ts:360]: Unnecessary async function

File: packages/beacon-node/src/chain/clock/LocalClock.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import {IBeaconClock} from "./interface.js";
 export class LocalClock implements IBeaconClock {
   private readonly config: IChainForkConfig;
   private readonly genesisTime: number;
-  private timeoutId: NodeJS.Timeout;
+  private timeoutId: number | NodeJS.Timeout;
```
Comment (dapplion) [inline packages/beacon-node/src/chain/clock/LocalClock.ts:17]: Allows to compile with any recent @types/node version. Has caused issues when installing individual local packages like c-kzg

File: packages/validator/test/unit/utils/params.test.ts [hunk 3]
```diff
@@ -12,20 +12,23 @@ const testCases: {name: string; items: [IChainConfig, Record<string, string>]}[]
...
+
 describe("utils / params / assertEqualParams", () => {
   it("default == default", () => {
     const chainConfigJson = chainConfigToJson(chainConfig);
     assertEqualParams(chainConfig, chainConfigJson);
   });
 
   it("default != other", () => {
-    const chainConfigJson = chainConfigToJson(chainConfig);
+    const ALTAIR_FORK_EPOCH = 10;
```
Comment (dapplion) [inline packages/validator/test/unit/utils/params.test.ts:24]: Make test deterministic test independent of current values of the default chain spec

**Review by twoeths — APPROVED** [review]

**Review by g11tech — APPROVED** [review]
lgtm

**Pull request merged**